### PR TITLE
Discorick/create issue on linked repo

### PR DESCRIPTION
**Pull request by discorick**

### CODE REVIEW

File: ember-app/app/controllers/issue/create.js [hunk 1]
```diff
@@ -31,9 +36,29 @@ var IssuesCreateController = Ember.ObjectController.extend({
...
+    return _.union([_.clone(board)], linked);
+  }.property("controllers.application.model.board.linkedRepos"),
+  issueIsLinked: function(){
+    return this.get("model.repo.full_name") !== this.get("controllers.application.model.board.full_name");
+  }.property("controllers.application.model.board.full_name", "model.repo.full_name"),
+  colorIssue: function(issue){
+    var linked_labels = this.get("controllers.application.model.board.link_labels");
+    var label = _.find(linked_labels, function(l){
+      var name = l.user + "/" + l.repo;
+      return name === issue.repo.full_name;
```
Comment (rauhryan) [inline ember-app/app/controllers/issue/create.js]: we need a case insensitive compare here

File: ember-app/app/controllers/issue/create.js [hunk 2]
```diff
@@ -5,14 +5,19 @@ var IssuesCreateController = Ember.ObjectController.extend({
   actions: {
     submit: function() {
       this.createIssue(this.get("order"));
+    },
+    assignRepo: function(repo){
+      this.set("model.repo.full_name", repo.full_name);
+      this.set("otherLabels", repo.other_labels);
+      this.set("model.labels", []);
+      this.set("milestones", repo.milestones);
+      this.set("assignees", repo.assignees);
```
Comment (rauhryan) [inline ember-app/app/controllers/issue/create.js]: Hmmm should we go the extra mile and "try" to match up the equivelent selections?

**Comment by rauhryan** [comment]
Aight, I'm done.. 

[Ready for review]

File: ember-app/app/templates/milestones/create.hbs [hunk 3]
```diff
@@ -20,7 +20,7 @@
         Due date: {{format-date dueDate}} 
         <small> <a href="#" {{action "clearDueDate"}}> clear </a> </small>
       </h5>
-      {{hb-date-picker dueDateBinding="model.due_on" processingBinding="processing"}}
+      {{hb-date-picker dueDate=model.due_on processing=processing }}
```
Comment (discorick) [inline ember-app/app/templates/milestones/create.hbs]: without the Binding on due_on the datepicker doesn't stay updated (at least it doesnt on ember 1.11.3)

File: ember-app/app/controllers/issue/create.js [hunk 4]
```diff
@@ -21,19 +22,52 @@ var IssuesCreateController = Ember.ObjectController.extend({
...
+    if (this.get("processing")){ 
+      return; 
+    }
     var controller = this;
     this.set("processing",true);
-    this.get("model").saveNew(order).then(function(issue){
+    this.get("model").save(order).then(function(issue){
+       if(controller.get("issueIsLinked")){
+         controller.colorIssue(issue);
+         controller.send("assignRepo", controller.get("controllers.application.model.board"));
```
Comment (discorick) [inline ember-app/app/controllers/issue/create.js]: We don't need to assignRepo if the component is already doing it on click do we ?

File: ember-app/app/controllers/issue/create.js [hunk 5]
```diff
@@ -21,19 +22,52 @@ var IssuesCreateController = Ember.ObjectController.extend({
...
+      var name = l.user + "/" + l.repo;
+      return name.toLowerCase() === issue.repo.full_name.toLowerCase();
+    });
+    issue.color = label.color;
+  },
+  actions: {
+    submit: function() {
+      this.createIssue(this.get("order"));
+    },
+    assignRepo: function(repo){
```
Comment (discorick) [inline ember-app/app/controllers/issue/create.js]: assign Repo actions no longer need to pass in an object

File: ember-app/app/templates/milestones/create.hbs (hunk 3 above)
Comment (rauhryan) [inline ember-app/app/templates/milestones/create.hbs]: notice they are unquoted

I think that may have been your issue, unquoted === bound property

File: ember-app/app/controllers/issue/create.js (hunk 4 above)
Comment (rauhryan) [inline ember-app/app/controllers/issue/create.js]: I left it there so the controller can reset the model properties for labels, assignee and milestone

File: ember-app/app/controllers/issue/create.js (hunk 5 above)
Comment (rauhryan) [inline ember-app/app/controllers/issue/create.js]: nice catch

**Comment by discorick** [comment]
Creating an issue on a linked repo using a milestone which doesn't exist on that linked repo sends the new issue to no-milestone. Is that acceptable ?

**Comment by rauhryan** [comment]
> Creating an issue on a linked repo using a milestone which doesn't exist on that linked repo sends the new issue to no-milestone. Is that acceptable ?

Hmmm, lets :shipit: and add an issue for later

**Comment by rauhryan** [comment]
rauhryan/huboard#557

**Pull request merged**